aptos-move/aptos-gas/src/gas_meter.rs

```diff
@@ -504,10 +505,13 @@ impl MoveGasMeter for StandardGasMeter {
                 )?;
             }
         }
+        if self.feature_version <= 8 && val.is_none() && bytes_loaded != 0.into() {
+            return Err(PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR).with_message("in legacy versions, number of bytes loaded must be zero when the resource does not exist ".to_string()));
```

Comment:
Should we just ignore these values instead of returning an error here? This would be a safer way to maintain backward incompatibility without just erroring out.

Reply:
I would actually argue the opposite: this check helps us ensure that changes I made elsewhere are indeed compatible with the old implementation, or otherwise this could fire during replay.